contracts/governance/extensions/GovernorCountingFractional.sol

```diff
@@ -0,0 +1,209 @@
+     * on the proposal, i.e. their checkpointed total weight minus votes already cast on the proposal.
+     *
+     * See `_countVoteNominal` and `_countVoteFractional` for more details.
+     */
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        uint256 totalWeight,
+        bytes memory params
+    ) internal virtual override returns (uint256) {
+        // Compute number of remaining votes. Returns 0 on overflow.
+        (, uint256 remainingWeight) = totalWeight.trySub(usedVotes(proposalId, account));
+        if (remainingWeight == 0) {
+            revert GovernorAlreadyCastVote(account);
+        }
+
+        if (params.length == 0) {
+            return _countVoteNominal(proposalId, account, support, remainingWeight);
+        } else if (params.length == 0x30) {
+            return _countVoteFractional(proposalId, account, params, remainingWeight);
+        } else {
+            revert GovernorInvalidVoteParams();
+        }
+    }
+
+    /**
+     * @dev Record votes with full weight cast for `support`.
+     *
+     * Because this function votes with the delegate's remaining weight, it can only be called once per proposal and
+     * thus does not require any replay protection.
+     */
+    function _countVoteNominal(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        uint256 weight
+    ) private returns (uint256) {
+        ProposalVote storage details = _proposalVotes[proposalId];
+        details.usedVotes[account] += weight;
+
+        if (support == uint8(GovernorCountingSimple.VoteType.Against)) {
+            details.againstVotes += weight;
+        } else if (support == uint8(GovernorCountingSimple.VoteType.For)) {
+            details.forVotes += weight;
+        } else if (support == uint8(GovernorCountingSimple.VoteType.Abstain)) {
+            details.abstainVotes += weight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
+
+        return weight;
+    }
+
+    /**
+     * @dev Count votes with fractional weight.
+     *
+     * `params` is expected to be tree packed uint128s:
+     * `abi.encodePacked(uint128(againstVotes), uint128(forVotes), uint128(abstainVotes))`
+     *
+     * This function can be called multiple times for the same account and proposal, i.e. partial/rolling votes are
+     * allowed. For example, an account with total weight of 10 could call this function three times with the
+     * following vote data:
+     *   - against: 1, for: 0, abstain: 2
+     *   - against: 3, for: 1, abstain: 0
+     *   - against: 1, for: 1, abstain: 1
+     * The result of these three calls would be that the account casts 5 votes AGAINST, 2 votes FOR, and 3 votes
+     * ABSTAIN on the proposal. Though partial, votes are still final once cast and cannot be changed or overridden.
+     * Subsequent partial votes simply increment existing totals.
```

Comment:
```suggestion
     * The `params` argument is expected to be three packed `uint128`:
     * `abi.encodePacked(uint128(againstVotes), uint128(forVotes), uint128(abstainVotes))`
     *
     * This function can be called multiple times for the same account and proposal (i.e. partial/rolling votes are
     * allowed). For example, an account with total weight of 10 could call this function three times with the
     * following vote data:
     *
     *   * Against: 1, For: 0, Abstain: 2
     *   * Against: 3, For: 1, Abstain: 0
     *   * Against: 1, For: 1, Abstain: 1
     *
     * Casting votes like this will make the calling account to cast a total of 5 `Against` votes, 2 `For` votes 
     * and 3 `Abstain` votes. Though partial, votes are still final once cast and cannot be changed or overridden.
     * Subsequent partial votes simply increment existing totals.
```